geonode/base/forms.py

```diff
@@ -337,16 +337,14 @@ def _define_choicefield(self, item, required, tname, lang):
 
     @staticmethod
     def _get_thesauro_keyword_label(item, lang):
-        qs_local = []
-        qs_non_local = [("", "------")]
-        for key in ThesaurusKeyword.objects.filter(thesaurus_id=item.id):
-            label = ThesaurusKeywordLabel.objects.filter(keyword=key).filter(lang=lang)
-            if label.exists():
-                qs_local.append((label.get().keyword.id, label.get().label))
-            else:
-                qs_non_local.append((key.id, key.alt_label))
+        keyword_id_for_given_thesaurus = ThesaurusKeyword.objects.filter(thesaurus_id=item)
+        qs_keyword_ids = ThesaurusKeywordLabel.objects.filter(lang=lang, keyword_id__in=keyword_id_for_given_thesaurus).values("keyword_id")
```

Comment:
Hi @mwallschlaeger 
It looks like there is some issue with changing the language. I'm trying it on master.
I'm trying it in Italian to see how it behaves. Setting in Italian, I have: `lang = it-it`,
![image](https://user-images.githubusercontent.com/51856725/182814280-9c077105-8360-4023-90b5-28ec52befe7f.png)

 so when it looks for the qs_keyword_ids, the queryset is always empty
![image](https://user-images.githubusercontent.com/51856725/182814575-68f5470b-43f4-4730-a326-0af39d7652fd.png)
 since it is saved with `lang=it`
![image](https://user-images.githubusercontent.com/51856725/182814459-d617ed1a-418a-4fe0-8539-0f5ecf8967c1.png)

So at the end qs_local is empty, 
![image](https://user-images.githubusercontent.com/51856725/182815171-d36747d1-7b36-42b3-ae7c-c0d467d9ad13.png)
while qs_non_local is full:
![image](https://user-images.githubusercontent.com/51856725/182815276-7e4dae6b-64dc-4d5e-9e32-8e56fe557dc1.png)